src/ion_data/mod.rs

```diff
@@ -1,4 +1,4 @@
-mod ion_eq;
+pub(crate) mod ion_eq;
```

Comment:
If you wanted, we could leave this mod the way it is, and you can use `IonData::eq(a, b)` to compare to values.

Reply:
Good call, I've done this in [158348e](https://github.com/amazon-ion/ion-rust/pull/674/commits/158348ea62837ceb94983db0863eca2de6105c25).